construct/core.py

```diff
@@ -524,10 +524,12 @@ def _parse(self, stream, context):
         except ConstructError:
             if c < self.mincount:
                 raise RangeError("expected %d to %d, found %d" %
-                    (self.mincount, self.maxcout, c), sys.exc_info()[1])
+                    (self.mincount, self.maxcout, c))
             stream.seek(pos)
         return obj
     def _build(self, obj, stream, context):
+        if (not hasattr(obj, '__len__')) or hasattr(obj, 'get'):
+            raise RangeError("expected sequence type, found %s" % type(obj))
         if len(obj) < self.mincount or len(obj) > self.maxcout:
```

Comment:
`if not slef.mincount <= len(obj) <= self.maxcout`
Either is correct. Which one do you like?